src/aerospike_clj/key.clj

```diff
@@ -0,0 +1,37 @@
+(ns aerospike-clj.key
+  (:import [com.aerospike.client Key Value]
+           [com.aerospike.client.util ThreadLocalData]))
+
+(defprotocol AerospikeIndex
+  (create-key-method ^Key [this ^String as-namesapce ^String set-name]))
+
+(extend-protocol AerospikeIndex
+  (Class/forName "[B") 
```

Comment:
I would add a comment `; byte-array`
Also maybe this should be put last as it requires an extra step to evaluate its java class

Reply:
added a comment. the evaluation happens at compile time.